logstash-core/lib/logstash/settings.rb

```diff
@@ -217,8 +217,58 @@ def initialize(name, default=nil, strict=true)
     end
 
     class Port < Setting
+      VALID_PORT_RANGE = 1..65535
+
+      def initialize(name, default=nil, strict=true)
+        super(name, ::Numeric, default, strict) { |value| valid?(value) }
+      end
+
+      def valid?(port)
+        VALID_PORT_RANGE.cover?(port)
+      end
+    end
+
+    class PortRange < Setting
+      PORT_SEPARATOR = "-"
+
       def initialize(name, default=nil, strict=true)
-        super(name, ::Numeric, default, strict) {|value| value >= 1 && value <= 65535 }
+        super(name, ::Range, coerce(default), strict=true)
+      end
+
+      def valid?(range)
+        Port::VALID_PORT_RANGE.first <= range.first && Port::VALID_PORT_RANGE.last >= range.last
+      end
+
+      def coerce(value)
```

Comment:
@jsvd I reviewed the #5773 and I will gladly rebase this PR on top of it with the coercible subclass.